### PR TITLE
[#384][Part 1] Increase Microsoft.Bot.Streaming code coverage

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses #384  (SW Repo)

## Description
This PR increases the code coverage for the `Microsoft.Bot.Streaming` from **63%** up to **84%**, the changes are divided in two parts, see PR #386 (SW Repo) for Part 2.

## Specific Changes
- Added unit tests for:
  - ProtocolAdapter
  - StreamExtensions
  - HeaderSerializer
  - PayloadAssembler
  - PayloadReceiver
  - PayloadSender
  - Stream

## Testing
The following image shows the before and after the changes that increased the code coverage.
imagen

### CODE REVIEW

**Comment by sw-joelmut** [comment]
Pipeline run: https://dev.azure.com/joelmut93/manx-pipeline-testing/_build/results?buildId=683&view=results

File: tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs [hunk 1]
```diff
@@ -2,7 +2,13 @@
 // Licensed under the MIT License.
 
 using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs:9]: This using is not needed.

File: tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs [hunk 2]
```diff
@@ -71,5 +90,309 @@ public void PayloadAssembler_Close__DoesNotSetEnd()
...
+            var streamManager = new StreamManager();
+
+            var header = new Header { Id = id, Type = PayloadTypes.Request };
+
+            static Task OnReceiveRequest(Guid id, ReceiveRequest req) => Task.CompletedTask;
+            static Task OnReceiveResponse(Guid id, ReceiveResponse res) => Task.CompletedTask;
+
+            var assembler = new PayloadAssemblerManager(streamManager, OnReceiveRequest, OnReceiveResponse);
+            var stream = assembler.GetPayloadStream(header);
+            stream = assembler.GetPayloadStream(header);
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs:158]: Is this ok? The stream variable is assigned twice.

File: tests/Microsoft.Bot.Streaming.Tests/StreamTests.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Streaming.Tests/StreamTests.cs:6]: These two usings are not needed.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs (hunk 2 above)
Comment (sw-joelmut) [reply tests/Microsoft.Bot.Streaming.Tests/Payloads/PayloadAssemblerTests.cs:158]: There is a condition inside the GetPayloadStream, that when there is an active assembler, it returns null, this process is just to cover that return line.
imagen

**Comment by sw-joelmut** [comment]
@ceciliaavila feedback applied, thanks

**Review by ceciliaavila — APPROVED** [review]

**Review by MartinLuccanera — APPROVED** [review]

**Review by GiulianoDolceTalianoSantoro — APPROVED** [review]

**Comment by ceciliaavila** [comment]
Promoted to MS in PR# 5839

**Pull request closed without merging**